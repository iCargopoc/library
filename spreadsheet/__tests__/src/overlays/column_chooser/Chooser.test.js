import React from "react";
import ReactDOM from "react-dom";
import ColumnReordering from "../../../../src/overlays/column_chooser/Chooser";
import ReactTestUtils, { act } from "react-dom/test-utils";
import { render, screen, fireEvent, getByRole } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { DndProvider } from "react-dnd";
import ColumnsList from "../../../../src/overlays/column_chooser/columnsList";

let container;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

test("chooser", () => {
    const props = {
        closeColumnReOrdering: jest.fn(),
        columns: [
            {
                key: "flightno",
                name: "FlightNo",
                draggable: false,
                editor: "Text",
                formulaApplicable: false,
                sortable: true,
                resizable: true,
                filterable: true,
                width: 150,
                filterType: "autoCompleteFilter"
            },
            {
                key: "date",
                name: "Date",
                draggable: false,
                editor: "DatePicker",
                formulaApplicable: false,
                sortable: true,
                resizable: true,
                filterable: true,
                width: 150,
                filterType: "autoCompleteFilter"
            },
            {
                key: "segmentfrom",
                name: "Segment From",
                draggable: false,
                editor: "DropDown",
                formulaApplicable: false,
                sortable: true,
                resizable: true,
                filterable: true,
                width: 150,
                filterType: "autoCompleteFilter"
            }
        ],
        headerKeys: ["FlightNo", "Date", "Segment From"],
        existingPinnedHeadersList: ["FlightNo", "Date", "Segment From"],
        maxLeftPinnedColumn: 5,
        updateTableAsPerRowChooser: jest.fn(),
        handleheaderNameList: jest.fn()
    };
    act(() => {
        ReactDOM.render(<ColumnReordering {...props} />, container);
    });
    let component = ReactDOM.render(<ColumnReordering {...props} />, container);
    component.resetColumnReorderList();
    component.componentWillUnmount();
    component.resetColumnReorderList();
    component.selectAllToColumnReOrderList();
    component.addToColumnReorderEntityList("Segment From");
    component.filterColumnReorderList({ target: { value: "FlightNo" } });
    component.filterColumnReorderList({ target: { value: "" } });
    component.reArrangeLeftPinnedColumn("Segment From");
    component.setWrapperRef("columns--grid");
    component.handleClickOutside({
        target: "parentDiv"
    });
});
test("render the chooser", () => {
    const columns = [
        {
            key: "flightno",
            name: "FlightNo",
            draggable: false,
            editor: "Text",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "date",
            name: "Date",
            draggable: false,
            editor: "DatePicker",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "segmentfrom",
            name: "Segment From",
            draggable: false,
            editor: "DropDown",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        }
    ];
    const maxLeftPinnedColumn = 5;
    const closeColumnReOrdering = jest.fn();
    const headerKeys = ["FlightNo", "Date", "Segment From"];
    const existingPinnedHeadersList = ["FlightNo", "Date", "Segment From"];
    const updateTableAsPerRowChooser = jest.fn();
    const handleheaderNameList = jest.fn();
    const { asFragment } = render(
        <ColumnReordering
            columns={columns}
            maxLeftPinnedColumn={maxLeftPinnedColumn}
            updateTableAsPerRowChooser={updateTableAsPerRowChooser}
            headerKeys={headerKeys}
            closeColumnReOrdering={closeColumnReOrdering}
            existingPinnedHeadersList={existingPinnedHeadersList}
            handleheaderNameList={handleheaderNameList}
        />
    );
    expect(
        asFragment(
            <ColumnReordering
                columns={columns}
                maxLeftPinnedColumn={maxLeftPinnedColumn}
                updateTableAsPerRowChooser={updateTableAsPerRowChooser}
                headerKeys={headerKeys}
                closeColumnReOrdering={closeColumnReOrdering}
                existingPinnedHeadersList={existingPinnedHeadersList}
                handleheaderNameList={handleheaderNameList}
            />
        )
    ).toMatchSnapshot();
});
it("reset button click event", () => {
    const columns = [
        {
            key: "flightno",
            name: "FlightNo",
            draggable: false,
            editor: "Text",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "date",
            name: "Date",
            draggable: false,
            editor: "DatePicker",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "segmentfrom",
            name: "Segment From",
            draggable: false,
            editor: "DropDown",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        }
    ];
    const maxLeftPinnedColumn = 5;
    const closeColumnReOrdering = jest.fn();
    const headerKeys = ["FlightNo", "Date", "Segment From"];
    const existingPinnedHeadersList = ["FlightNo", "Date", "Segment From"];
    const updateTableAsPerRowChooser = jest.fn();
    const handleheaderNameList = jest.fn();
    const { getByTestId } = render(
        <ColumnReordering
            columns={columns}
            maxLeftPinnedColumn={maxLeftPinnedColumn}
            updateTableAsPerRowChooser={updateTableAsPerRowChooser}
            headerKeys={headerKeys}
            closeColumnReOrdering={closeColumnReOrdering}
            existingPinnedHeadersList={existingPinnedHeadersList}
            handleheaderNameList={handleheaderNameList}
        />
    );
    fireEvent.click(getByTestId("resetButton"));
});
it("cancel button click event", () => {
    const columns = [
        {
            key: "flightno",
            name: "FlightNo",
            draggable: false,
            editor: "Text",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "date",
            name: "Date",
            draggable: false,
            editor: "DatePicker",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "segmentfrom",
            name: "Segment From",
            draggable: false,
            editor: "DropDown",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        }
    ];
    const maxLeftPinnedColumn = 5;
    const closeColumnReOrdering = jest.fn();
    const headerKeys = ["FlightNo", "Date", "Segment From"];
    const existingPinnedHeadersList = ["FlightNo", "Date", "Segment From"];
    const updateTableAsPerRowChooser = jest.fn();
    const handleheaderNameList = jest.fn();
    const { getByTestId } = render(
        <ColumnReordering
            columns={columns}
            maxLeftPinnedColumn={maxLeftPinnedColumn}
            updateTableAsPerRowChooser={updateTableAsPerRowChooser}
            headerKeys={headerKeys}
            closeColumnReOrdering={closeColumnReOrdering}
            existingPinnedHeadersList={existingPinnedHeadersList}
            handleheaderNameList={handleheaderNameList}
        />
    );
    fireEvent.click(getByTestId("cancelButton"));
});
it("save button click event", () => {
    const columns = [
        {
            key: "flightno",
            name: "FlightNo",
            draggable: false,
            editor: "Text",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "date",
            name: "Date",
            draggable: false,
            editor: "DatePicker",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "segmentfrom",
            name: "Segment From",
            draggable: false,
            editor: "DropDown",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        }
    ];
    const maxLeftPinnedColumn = 5;
    const closeColumnReOrdering = jest.fn();
    const headerKeys = ["FlightNo", "Date", "Segment From"];
    const existingPinnedHeadersList = ["FlightNo", "Date", "Segment From"];
    const updateTableAsPerRowChooser = jest.fn();
    const handleheaderNameList = jest.fn();
    const { getByTestId } = render(
        <ColumnReordering
            columns={columns}
            maxLeftPinnedColumn={maxLeftPinnedColumn}
            updateTableAsPerRowChooser={updateTableAsPerRowChooser}
            headerKeys={headerKeys}
            closeColumnReOrdering={closeColumnReOrdering}
            existingPinnedHeadersList={existingPinnedHeadersList}
            handleheaderNameList={handleheaderNameList}
        />
    );
    fireEvent.click(getByTestId("saveButton"));
});
it("checkBox change event", () => {
    const columns = [
        {
            key: "flightno",
            name: "FlightNo",
            draggable: false,
            editor: "Text",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "date",
            name: "Date",
            draggable: false,
            editor: "DatePicker",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "segmentfrom",
            name: "Segment From",
            draggable: false,
            editor: "DropDown",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        }
    ];
    const maxLeftPinnedColumn = 5;
    const closeColumnReOrdering = jest.fn();
    const headerKeys = ["FlightNo", "Date", "Segment From"];
    const existingPinnedHeadersList = ["FlightNo", "Date", "Segment From"];
    const updateTableAsPerRowChooser = jest.fn();
    const handleheaderNameList = jest.fn();
    const { getAllByTestId } = render(
        <ColumnReordering
            columns={columns}
            maxLeftPinnedColumn={maxLeftPinnedColumn}
            updateTableAsPerRowChooser={updateTableAsPerRowChooser}
            headerKeys={headerKeys}
            closeColumnReOrdering={closeColumnReOrdering}
            existingPinnedHeadersList={existingPinnedHeadersList}
            handleheaderNameList={handleheaderNameList}
        />
    );
    expect(getAllByTestId("checkBox")).toHaveLength(3);
    fireEvent.change(getAllByTestId("checkBox")[0], {
        target: { checked: true }
    });
    fireEvent.change(getAllByTestId("checkBox")[0], {
        target: { checked: false }
    });
});
it("select All Columns checkBox change event", () => {
    const columns = [
        {
            key: "flightno",
            name: "FlightNo",
            draggable: false,
            editor: "Text",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "date",
            name: "Date",
            draggable: false,
            editor: "DatePicker",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        },
        {
            key: "segmentfrom",
            name: "Segment From",
            draggable: false,
            editor: "DropDown",
            formulaApplicable: false,
            sortable: true,
            resizable: true,
            filterable: true,
            width: 150,
            filterType: "autoCompleteFilter"
        }
    ];
    const maxLeftPinnedColumn = 5;
    const closeColumnReOrdering = jest.fn();
    const headerKeys = ["FlightNo", "Date", "Segment From"];
    const existingPinnedHeadersList = ["FlightNo", "Date", "Segment From"];
    const updateTableAsPerRowChooser = jest.fn();
    const handleheaderNameList = jest.fn();
    const { getAllByTestId } = render(
        <ColumnReordering
            columns={columns}
            maxLeftPinnedColumn={maxLeftPinnedColumn}
            updateTableAsPerRowChooser={updateTableAsPerRowChooser}
            headerKeys={headerKeys}
            closeColumnReOrdering={closeColumnReOrdering}
            existingPinnedHeadersList={existingPinnedHeadersList}
            handleheaderNameList={handleheaderNameList}
        />
    );
    expect(getAllByTestId("selectAllColumns")).toHaveLength(1);
    fireEvent.change(getAllByTestId("selectAllColumns")[0], {
        target: { checked: true }
    });
    fireEvent.change(getAllByTestId("selectAllColumns")[0], {
        target: { checked: false }
    });
});
