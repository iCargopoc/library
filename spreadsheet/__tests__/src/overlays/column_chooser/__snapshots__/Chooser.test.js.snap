// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the chooser 1`] = `
<DocumentFragment>
  <div
    class="neo-popover neo-popover--column columns--grid"
  >
    <div
      class="neo-popover__column column__grid"
    >
      <div
        class="column__chooser"
      >
        <div
          class="column__header"
        >
          <div
            class=""
          >
            <strong>
              Column Chooser
            </strong>
          </div>
        </div>
        <div
          class="column__body"
        >
          <div>
            <input
              class="custom__ctrl"
              placeholder="Search column"
              type="text"
            />
          </div>
          <div
            class="column__selectAll"
          >
            <div
              class="column__checkbox"
            >
              <input
                checked=""
                data-testid="selectAllColumns"
                id="selectallcolumncheckbox"
                type="checkbox"
              />
            </div>
            <div
              class="column__txt"
            >
              Select all
            </div>
          </div>
          <div
            class="column__wrap"
          >
            <div
              class="column__checkbox"
            >
              <input
                checked=""
                data-testid="checkBox"
                id="checkboxtoselectreorder_FlightNo"
                type="checkbox"
              />
            </div>
            <div
              class="column__txt"
            >
              FlightNo
            </div>
          </div>
          <div
            class="column__wrap"
          >
            <div
              class="column__checkbox"
            >
              <input
                checked=""
                data-testid="checkBox"
                id="checkboxtoselectreorder_Date"
                type="checkbox"
              />
            </div>
            <div
              class="column__txt"
            >
              Date
            </div>
          </div>
          <div
            class="column__wrap"
          >
            <div
              class="column__checkbox"
            >
              <input
                checked=""
                data-testid="checkBox"
                id="checkboxtoselectreorder_Segment From"
                type="checkbox"
              />
            </div>
            <div
              class="column__txt"
            >
              Segment From
            </div>
          </div>
        </div>
      </div>
      <div
        class="column__settings"
      >
        <div
          class="column__header"
        >
          <div
            class="column__headerTxt"
          >
            <strong>
              Column Settings
            </strong>
          </div>
          <div
            class="column__close"
          >
            <i>
              <img
                alt="Column chooser Close Icon"
                src="[object Object]"
              />
            </i>
          </div>
        </div>
        <div
          class="column__body"
        >
          <div
            class="column__info"
          >
            <strong>
                  Selected Column Count : 3
            </strong>
            <strong>
                  Left Pinned Column Count Remaining : 2
            </strong>
          </div>
          <div
            data-testid="columnItem"
            style="display: flex; flex-wrap: wrap;"
          >
            <div
              style="cursor: move; opacity: 1;"
            >
              <div
                class="column__reorder"
              >
                <div
                  class="column_drag"
                  style="cursor: move;"
                >
                  <i>
                    <img
                      alt="Column Chooser Drag Icon"
                      src="[object Object]"
                    />
                  </i>
                </div>
                <div
                  class="column__reorder__name"
                >
                  FlightNo
                </div>
                <div
                  class="column__innerCells__wrap"
                >
                  <div
                    class="column__wrap"
                  >
                    <div
                      class="column__checkbox"
                    >
                      <input
                        checked=""
                        id="checkBoxToPinLeft_FlightNo"
                        role="button"
                        type="checkbox"
                      />
                    </div>
                    <div
                      class="column__txt"
                    >
                      Pin Left
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="cursor: move; opacity: 1;"
            >
              <div
                class="column__reorder"
              >
                <div
                  class="column_drag"
                  style="cursor: move;"
                >
                  <i>
                    <img
                      alt="Column Chooser Drag Icon"
                      src="[object Object]"
                    />
                  </i>
                </div>
                <div
                  class="column__reorder__name"
                >
                  Date
                </div>
                <div
                  class="column__innerCells__wrap"
                >
                  <div
                    class="column__wrap"
                  >
                    <div
                      class="column__checkbox"
                    >
                      <input
                        checked=""
                        id="checkBoxToPinLeft_Date"
                        role="button"
                        type="checkbox"
                      />
                    </div>
                    <div
                      class="column__txt"
                    >
                      Pin Left
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="cursor: move; opacity: 1;"
            >
              <div
                class="column__reorder"
              >
                <div
                  class="column_drag"
                  style="cursor: move;"
                >
                  <i>
                    <img
                      alt="Column Chooser Drag Icon"
                      src="[object Object]"
                    />
                  </i>
                </div>
                <div
                  class="column__reorder__name"
                >
                  Segment From
                </div>
                <div
                  class="column__innerCells__wrap"
                >
                  <div
                    class="column__wrap"
                  >
                    <div
                      class="column__checkbox"
                    >
                      <input
                        checked=""
                        id="checkBoxToPinLeft_Segment From"
                        role="button"
                        type="checkbox"
                      />
                    </div>
                    <div
                      class="column__txt"
                    >
                      Pin Left
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="column__footer"
        >
          <div
            class="column__btns"
          >
            <button
              class="btns"
              data-testid="resetButton"
              type="button"
            >
              Reset
            </button>
            <button
              class="btns"
              data-testid="cancelButton"
              type="button"
            >
              Cancel
            </button>
            <button
              class="btns btns__save"
              data-testid="saveButton"
              type="button"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
